pkg/operator/controller/nodeobservabilityrun/controller.go

```diff
@@ -87,23 +87,34 @@ func (r *NodeObservabilityRunReconciler) Reconcile(ctx context.Context, req ctrl
 		return
 	}
 
+	defer func() {
+		errUpdate := r.Status().Update(ctx, instance, &client.UpdateOptions{})
+		if errUpdate != nil {
+			r.Log.Error(err, "failed to update status")
+			err = utilerrors.NewAggregate([]error{err, errUpdate})
+		}
+	}()
+
 	// interact with MCO only if we are executing CrioKubeletProfile runs
 	// also bypass if the runtype is e2e-test
 	if instance.Spec.RunType != nodeobservabilityv1alpha1.E2ETest && (instance.Spec.RunType == nodeobservabilityv1alpha1.CrioKubeletProfile) {
```

Comment:
I'm strongly against volkswagen style testing. System Under Test should be identical to the system that will run in production env.